TYPO3.Flow/Classes/TYPO3/Flow/Command/ResourceCommandController.php

```diff
@@ -84,7 +84,10 @@ public function publishCommand($collection = null)
             foreach ($collections as $collection) {
                 /** @var CollectionInterface $collection */
                 $this->outputLine('Publishing resources of collection "%s"', array($collection->getName()));
-                $collection->publish();
+                $target = $collection->getTarget();
```

Comment:
Clearing the state, need to be done carefully, all object are detatched in doctrine, during a web request it's pretty dangereous (and even impossible currently). Check the code of the resource clean command, to see how I restore object to do the cleanup.